lib/auth/grpcserver.go

```diff
@@ -2533,7 +2533,9 @@ func isLocalProxyCertReq(req *authpb.UserCertsRequest) bool {
 	return (req.Usage == authpb.UserCertsRequest_Database &&
 		req.RequesterName == authpb.UserCertsRequest_TSH_DB_LOCAL_PROXY_TUNNEL) ||
 		(req.Usage == authpb.UserCertsRequest_Kubernetes &&
-			req.RequesterName == authpb.UserCertsRequest_TSH_KUBE_LOCAL_PROXY)
+			req.RequesterName == authpb.UserCertsRequest_TSH_KUBE_LOCAL_PROXY) ||
+		(req.Usage == authpb.UserCertsRequest_App &&
+			req.RequesterName == authpb.UserCertsRequest_TSH_APP_LOCAL_PROXY)
```

Comment:
I forgot to rebuild `teleport` and ran `tsh proxy app` against a cluster that didn't include this change.

`tsh proxy app` would ask for a tap and then exit silently. Re-running it in debug mode shows that an error was returned ("single-use certificates cannot be issued for app access").

Could you make sure that the error is surfaced to the user? This should help end users if starting a proxy fails for some other reason.

<details>
<summary>Error log from tsh app proxy -d</summary>

```
Tap any security key
Detected security key tap
2024-04-25T17:30:23+02:00 DEBU [CLIENT]    Issuing single-use certificate from unary GenerateUserCerts client/cluster_client.go:609
2024-04-25T17:30:23+02:00 ERRO [TSH]       Failed to start local proxy. error:[
ERROR REPORT:
Original Error: *interceptors.RemoteError single-use certificates cannot be issued for app access
Stack Trace:
	github.com/gravitational/teleport/api@v0.0.0/client/client.go:1102 github.com/gravitational/teleport/api/client.(*Client).GenerateUserCerts
	github.com/gravitational/teleport/lib/client/cluster_client.go:610 github.com/gravitational/teleport/lib/client.PerformMFACeremony
	github.com/gravitational/teleport/lib/client/cluster_client.go:373 github.com/gravitational/teleport/lib/client.(*ClusterClient).performMFACeremony
	github.com/gravitational/teleport/lib/client/cluster_client.go:486 github.com/gravitational/teleport/lib/client.(*ClusterClient).IssueUserCertsWithMFA
	github.com/gravitational/teleport/lib/client/api.go:1427 github.com/gravitational/teleport/lib/client.(*TeleportClient).IssueUserCertsWithMFA
	github.com/gravitational/teleport/lib/client/local_proxy_middleware.go:181 github.com/gravitational/teleport/lib/client.(*appCertReissuer).ReissueCert.func1
	github.com/gravitational/teleport/lib/client/api.go:592 github.com/gravitational/teleport/lib/client.RetryWithRelogin
	github.com/gravitational/teleport/lib/client/local_proxy_middleware.go:180 github.com/gravitational/teleport/lib/client.(*appCertReissuer).ReissueCert
	github.com/gravitational/teleport/lib/client/local_proxy_middleware.go:90 github.com/gravitational/teleport/lib/client.(*CertChecker).ensureValidCerts
	github.com/gravitational/teleport/lib/client/local_proxy_middleware.go:79 github.com/gravitational/teleport/lib/client.(*CertChecker).OnStart
	github.com/gravitational/teleport/lib/srv/alpnproxy/local_proxy.go:177 github.com/gravitational/teleport/lib/srv/alpnproxy.(*LocalProxy).start
	github.com/gravitational/teleport/lib/srv/alpnproxy/local_proxy.go:171 github.com/gravitational/teleport/lib/srv/alpnproxy.(*LocalProxy).Start
	github.com/gravitational/teleport/tool/tsh/common/proxy.go:387 github.com/gravitational/teleport/tool/tsh/common.onProxyCommandApp
	github.com/gravitational/teleport/tool/tsh/common/tsh.go:1454 github.com/gravitational/teleport/tool/tsh/common.Run
	github.com/gravitational/teleport/tool/tsh/common/tsh.go:594 github.com/gravitational/teleport/tool/tsh/common.Main
	github.com/gravitational/teleport/tool/tsh/main.go:26 main.main
	runtime/proc.go:271 runtime.main
	runtime/asm_arm64.s:1222 runtime.goexit
User Message: single-use certificates cannot be issued for app access] common/proxy.go:388
```
</details>